### ISSUE TITLE
compileAndRender and compileToFunction

### ISSUE THREAD

**Issue by bentomas** [akdubya/dustjs#7]
It would be nice if we didn't have to compile templates to a name, and could just say 

```
dust.compileAndRender(fs.readFileSync(...), { ... }, function(rendered) { ... });
```

Or if we could compile the template to a function that just takes data.  

```
var func = dust.compileToFunction(fs.readFileSync(...));
func({...}, ...);
func({...}, ...);
func({...}, ...);
// and so on
```

I don't have a need for the named templates stuff, but still think dust is really cool.  I do a lot of templating with one off templates that aren't going to be used again, and having a more flexible way of using Dust would be awesome!

**Comment by semanticprogrammer** [comment]
I like this idea. Please implement it.

**Comment by akdubya** [comment]
I'm still thinking about the API, but this seems reasonable enough.

**Comment by akdubya** [comment]
added dust.renderSource and dust.compileFn (closed by c5302a135aedd4be4e7876921af17bb6abb8dc3b)

dust.renderSource(&#39;hello&#39;, {}, fn) --&gt; render with callback
dust.renderSource(&#39;hello&#39;, {}).on(&#39;data&#39;, fn) --&gt; render as stream
dust.compileFn(&#39;hello&#39;) --&gt; function(context, callback) { ... }

**Comment by bentomas** [comment]
Wow!  That was quick!  Awesome, trying it out now.
